bigquery/storage/managedwriter/send_optimizer.go

```diff
@@ -104,32 +107,47 @@ func (mo *multiplexOptimizer) optimizeSend(arc storagepb.BigQueryWrite_AppendRow
 			swapOnSuccess = true
 		} else {
 			// same stream
-			if !proto.Equal(mo.prev.GetProtoRows().GetWriterSchema().GetProtoDescriptor(), cp.GetProtoRows().GetWriterSchema().GetProtoDescriptor()) {
+			if !proto.Equal(getDescriptorFromAppend(mo.prev), getDescriptorFromAppend(cp)) {
 				swapOnSuccess = true
 			} else {
 				// the redaction case, where we won't swap.
-				cp.GetProtoRows().WriterSchema = nil
+				if pr := cp.GetProtoRows(); pr != nil {
+					pr.WriterSchema = nil
+				}
 			}
 		}
-		resp = arc.Send(cp)
-		if resp == nil && swapOnSuccess {
+		err = arc.Send(cp)
+		if err == nil && swapOnSuccess {
 			cp.GetProtoRows().Rows = nil
```

Comment:
might be good to add a check to see if cp.GetProtoRows() returns nil like was added on other cases

Reply:
Good spot, updating.